utils/snapshot_util.py

```diff
@@ -23,56 +27,143 @@ def load_snapshot(self, snapshot_file_path):
+            return False, new_violation_details
+
+        # If no difference is found
+        return True, []
+
+    def extract_diff_details(self, violation_diff, page):
+        "Extract details from the violation diff."
+        violation_details = []
+
+        # Handle newly added violations (dictionary_item_added)
+        for key, value in violation_diff.get('dictionary_item_added', {}).items():
+            violation_details.append({
+                "page": f"{page}- New Item added",
+                "id": value['id'],
+                "key": key,
+                "impact": value.get('impact', 'Unknown'),
+                "description": value.get('description', 'Unknown'),
+                "nodes": value.get('nodes', 'Unknown')
+            })
+
+        # Handle removed violations (dictionary_item_removed)
+        for key, value in violation_diff.get('dictionary_item_removed', {}).items():
+            violation_details.append({
+                "page": page,
+                "id": value['id'],
+                "key": key,
+                "impact": value.get('impact', 'Unknown'),
+                "description": value.get('description', 'Unknown'),
+                "nodes": value.get('nodes', 'Unknown')
+            })
+
+        # Handle changes to existing violations (values_changed)
+        for key, value in violation_diff.get('values_changed', {}).items():
+            path = key.split("']")[-2]
+            old_value = value['old_value']
+            new_value = value['new_value']
+            violation_details.append({
+                "page": page,
+                "id": path,
+                "key": key,
+                "impact": value.get('new_value', 'Unknown'),
+                "description": f"Changed from- {old_value} \n\t\t To- {new_value}",
+                "nodes": key
+            })
+
+        # Handle added items in iterable (iterable_item_added)
+        for key, value in violation_diff.get('iterable_item_added', {}).items():
+            violation_details.append({
+                "page": page,
+                "id": value.get('id', 'Unknown'),
+                "key": key,
+                "impact": value.get('impact', 'Unknown'),
+                "description": f"New node item added: {key}",
+                "nodes": str(value)
+            })
+
+        # Handle removed items in iterable (iterable_item_removed)
+        for key, value in violation_diff.get('iterable_item_removed', {}).items():
+            violation_details.append({
+                "page": page,
+                "id": value.get('id', 'Unknown'),
+                "key": key,
+                "impact": value.get('impact', 'Unknown'),
+                "description": f"Item node removed: {key}",
+                "nodes": str(value)
+            })
+
+        return violation_details
+
+    def log_new_violations(self, new_violation_details, test_obj):
+        "Log details of new violations to the console."
```

Comment:
@mohanqxf2 Do we have some other way to write messages to log file. I don't think passing test_obj is the right way here